tokio/src/runtime/blocking/shutdown.rs

```diff
@@ -9,11 +9,13 @@ use crate::sync::oneshot;
 use std::time::Duration;
 
 #[derive(Debug, Clone)]
+#[repr(C)]
 pub(super) struct Sender {
     _tx: Arc<oneshot::Sender<()>>,
 }
```

Comment:
For example, here you have a field of type `std::sync::Arc`, but the standard library does not put `#[repr(C)]` on that type.

Reply:
yessss...i'm soo wrong...
Ok, I'm starting to think that it's impossible due to the missing ABI rust nature...

Reply:
It's possible that there is some way to solve it by creating an interface based on C FFI to translate things as they are moved across the boundary, but it would be a complicated venture and (unlike what you did with this PR) not something you should start spending time on without first discussing the details of the approach with us.